services/configuration_repository/configuration_repository_decorator.rb

```diff
@@ -0,0 +1,31 @@
+require_relative "../services"
+
+module FastlaneCI
+  # This module acts as a decorator to be added to any method that potentially will make changes
+  # into the ci-config repository directory.
+  module ConfigurationRepositoryUpdater
+    # This decorator method is responsible of pulling changes from the ci-config repo.
+    # rubocop:disable Style/ClassVars
+    @@mutex = Mutex.new
```

Comment:
`mutex` for what? Let's make a better name 👍

Reply:
I'm terrible at naming, any suggestion?

Reply:
Whatever the mutex is protecting + `_` ha.